interop_binaries/tests/end_to_end.rs

```diff
@@ -8,173 +12,87 @@ use janus_server::task::PRIO3_AES128_VERIFY_KEY_LENGTH;
-            Err(e) if e.kind() == ErrorKind::InvalidInput => {}
-            Err(e) => panic!("failed to kill child process: {:?}", e),
-        }
-    }
-}
-
-/// Pass output from a child process's stdout pipe to print!(), so that it can be captured and
-/// stored by the test harness.
-async fn forward_stdout(stdout: ChildStdout) -> io::Result<()> {
-    let mut reader = BufReader::new(stdout);
-    let mut line = String::new();
-    loop {
-        line.clear();
-        let count = reader.read_line(&mut line).await?;
-        if count == 0 {
-            return Ok(());
-        }
-        print!("{}", line);
-    }
-}
-
-/// Pass output from a child process's stderr pipe to eprint!(), so that it can be captured and
-/// stored by the test harness.
-async fn forward_stderr(stderr: ChildStderr) -> io::Result<()> {
-    let mut reader = BufReader::new(stderr);
-    let mut line = String::new();
-    loop {
-        line.clear();
-        let count = reader.read_line(&mut line).await?;
-        if count == 0 {
-            return Ok(());
-        }
-        eprint!("{}", line);
-    }
-}
-
 /// Take a VDAF description and a list of measurements, perform an entire aggregation using
 /// interoperation test binaries, and return the aggregate result. This follows the outline of
 /// section 4.7 of draft-dcook-ppm-dap-interop-test-design-00.
 async fn run(
     vdaf_object: serde_json::Value,
     measurements: &[serde_json::Value],
     aggregation_parameter: &[u8],
-) -> anyhow::Result<serde_json::Value> {
-    // Start up a database testcontainer for each aggregator directly, and don't set up the schema.
-    let leader_db_container =
-        CONTAINER_CLIENT.run(RunnableImage::from(Postgres::default()).with_tag("14-alpine"));
-    let leader_postgres_port = leader_db_container.get_host_port_ipv4(5432);
-    let helper_db_container =
-        CONTAINER_CLIENT.run(RunnableImage::from(Postgres::default()).with_tag("14-alpine"));
-    let helper_postgres_port = helper_db_container.get_host_port_ipv4(5432);
-
-    // Pick four ports for HTTP servers.
-    let client_port = pick_unused_port().context("couldn't pick a port for the client")?;
-    let leader_port = pick_unused_port().context("couldn't pick a port for the leader")?;
-    let helper_port = pick_unused_port().context("couldn't pick a port for the helper")?;
-    let collector_port = pick_unused_port().context("couldn't pick a port for the collector")?;
-    assert_eq!(
-        BTreeSet::from([client_port, leader_port, helper_port, collector_port]).len(),
-        4,
-        "Ports selected for HTTP servers were not unique",
-    );
-
-    // Create and start containers. (here, we just run the binaries instead)
+) -> serde_json::Value {
+    // Create and start containers.
     // We use std::process instead of tokio::process so that we can kill the child processes from
     // a Drop implementation. tokio::process::Child::kill() is async, and could not be called from
     // there.
```

Comment:
This part of the comment is obsolete now.
```suggestion
```

Reply:
👍🏻 (btw, this is entirely moot with this change & even before this change, using a stdlib non-async process was fine, but you can work around the technical issue of not being able to run async code in a Drop implementation by using a "shutdown" signal that the Drop implementation signals, along with a "shutdown complete" signal that the Drop implementation then waits on; see [Daphne's drop implementation](https://github.com/divviup/janus/blob/e842b61f0527843c4edba44881ade8196cd7594c/monolithic_integration_test/src/daphne.rs#L226-L233) for an example)